tls/extensions/s2n_client_server_name.h

```diff
@@ -20,3 +20,4 @@
 #include "tls/s2n_connection.h"
 
 extern const s2n_extension_type s2n_client_server_name_extension;
+S2N_RESULT s2n_parse_server_name(struct s2n_stuffer *extension, struct s2n_blob *server_name);
```

Comment:
Naming nit: "s2n_server_name_parse" instead, or maybe even "s2n_client_server_name_parse". I was very sad recently that "s2n_wipe_record" looked funny when I moved it to the record-related headers :)